app/models/division_parameter_parser.rb

```diff
@@ -0,0 +1,48 @@
+class DivisionParameterParser
+
+  YEAR_ONLY_REGEX = /^\d{4}$/
+  YEAR_AND_MONTH_REGEX = /^\d{4}-\d{1,2}$/
+  COMPLETE_DATE_REGEX = /^\d{4}-\d{1,2}-\d{1,2}$/
+
+  def self.get_date_range(date)
+
+    if date =~ YEAR_ONLY_REGEX
+      return get_year_range(date)
+    elsif date =~ YEAR_AND_MONTH_REGEX
+      return get_month_range(date)
+    elsif date =~ COMPLETE_DATE_REGEX
+      return get_day_range(date)
+    end
+
+    raise ArgumentError, 'Not a valid date format'
+  end
+
+private
+
+  def self.get_year_range(year)
+    date = Date.parse("#{year}-01-01")
```

Comment:
I think semantically makes sense, because that's exactly what we are trying to do: parse a Date. Also, it seems the `.new` method does not accept a `String` as a parameter.

http://ruby-doc.org/stdlib-2.4.0/libdoc/date/rdoc/Date.html#method-c-new